File: Calcolatrice/App/app/src/main/java/com/example/samue/calcolatrice/MainActivity.java
package com.example.samue.calcolatrice;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;
import android.widget.GridView;
import android.widget.ListAdapter;
import android.widget.ArrayAdapter;
import android.widget.AdapterView;
import android.view.View;
import static java.lang.Integer.parseInt;
import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    GridView Calculator;
    TextView Res;
    String operator;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Calculator = (GridView)findViewById(R.id.Buttons);
        Res = (TextView) findViewById(R.id.Cont);

        String[] numbers = new String[]{"1", "2", "3", "CE", "4", "5", "6", "+", "7", "8", "9", "-", "=", "0", "*", "/" };

        ListAdapter adapter = new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, numbers);
        Calculator.setAdapter(adapter);

        Calculator.setOnItemClickListener(new AdapterView.OnItemClickListener()
        {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                String Selection = parent.getItemAtPosition(position).toString();

                if(!Selection.equals("+") && !Selection.equals("-") && !Selection.equals("*") && !Selection.equals("/") && !Selection.equals("CE") && !Selection.equals("="))
                {
                    if(Res.getText().equals(""))
                    {
                        Res.setText(Selection);
                    }
                    else
                    {
                        Res.setText(Res.getText() + Selection);
                    }
                }
                else
                {
                    float result = 0;
                    switch (Selection)
                    {
                        case "CE":
                            result = 0;
                            Res.setText("");
                            break;
                        case "+":
                            operator = Selection;
                            Res.setText(Res.getText() + Selection);
                            break;
                        case "-":
                            operator = Selection;
                            Res.setText(Res.getText() + Selection);
                            break;
                        case "*":
                            operator = Selection;
                            Res.setText(Res.getText() + Selection);
                            break;
                        case "/":
                            operator = Selection;
                            Res.setText(Res.getText() + Selection);
                            break;
                        case "=":
                            switch(operator)
                            {
                                case "+":
                                    result = parseInt("" + Res.getText().charAt(0)) + parseInt("" + Res.getText().charAt(Res.getText().length() - 1));
                                    break;
                                case "-":
                                    result = parseInt("" + Res.getText().charAt(0)) - parseInt("" + Res.getText().charAt(Res.getText().length() - 1));
                                    break;
                                case "*":
                                    result = parseInt("" + Res.getText().charAt(0)) * parseInt("" + Res.getText().charAt(Res.getText().length() - 1));
                                    break;
                                case "/":
                                    result = parseInt("" + Res.getText().charAt(0)) / parseInt("" + Res.getText().charAt(Res.getText().length() - 1));
                                    break;
                            }
                            Res.setText(String.valueOf(result));
                            result = 0;
                            break;
                    }
                }
            }
        });
    }
}

File: 290118_Calculator/app/src/main/java/com/example/filippofantinato/calculator/MainActivity.kt
package com.example.filippofantinato.calculator

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.TextView
import java.util.*

class MainActivity : AppCompatActivity()
{

    override fun onCreate(savedInstanceState: Bundle?)
    {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        var error: Boolean=false
        val buttonsId: IntArray = intArrayOf(R.id.one, R.id.two, R.id.three,
                    R.id.four, R.id.five, R.id.six,
                    R.id.seven, R.id.eight, R.id.nine,
                    R.id.dot, R.id.zero, R.id.plus, R.id.minus,
                    R.id.mol, R.id.div)
        val inputText: TextView = findViewById<TextView>(R.id.inputText)
        val delete: Button = findViewById(R.id.delete)
        val equal: Button = findViewById(R.id.equal)
        delete.setOnClickListener({view ->
            inputText.text=""
        })
        equal.setOnClickListener({view ->
            val scanner = Scanner(inputText.text.toString())
            Log.d("Main", inputText.text.toString())
            try{
                val op1: Int = scanner.nextInt()
                val op2: String = scanner.next()
                val op3: Int = scanner.nextInt()
                var output: Int? = null
                when(op2)
                {
                    "+" -> {
                        output=op1+op3
                    }
                    "-" -> {
                        output=op1-op3
                    }
                    "*" -> {
                        output=op1*op3
                    }
                    "/" -> {
                        output=op1/op3
                    }
                }
                inputText.text=output.toString()
            }catch(e: InputMismatchException){
                inputText.text="Error"
                error=true
            }
        })
        for(buttonId in buttonsId)
        {
            findViewById<Button>(buttonId).setOnClickListener({view ->
                var actualText: String
                val pressedText: String = (view as Button).text.toString()
                var space: String = if (pressedText.equals("+")||pressedText.equals("-")||pressedText.equals("*")||pressedText.equals("/")) " " else ""
                if(error)
                {
                    actualText=""
                    error=false
                }else{
                    actualText=inputText.text.toString()
                }
                inputText.text=actualText+space+pressedText+space
            })
        }
    }
}
